### ISSUE TITLE
An error occured in the receive middleware:  TypeError: Cannot read property 'text' of undefined

### ISSUE THREAD

**Issue by Abgaryan** [howdyai/botkit-starter-web#33]
Error Log :
```
 An error occured in the receive middleware:  TypeError: Cannot read property 'text' of undefined
  at /usr/src/app/node_modules/botkit/lib/CoreBot.js:256:49
```
The response  I am getting in default callback : 

`type=message_received, text=Italian, user=290e8ba3-785c-db03-52b8-8cd1e290a19f, channel=socket, type=message, text=Italian, user=290e8ba3-785c-db03-52b8-8cd1e290a19f, channel=socket, stage=receive, question=What’s your nationality?`

**Comment by peterswimm** [comment]
Can you post a snippet of the code that creates this problem?

**Comment by Abgaryan** [comment]
```
let commonNationalities = [
    {
      "content_type": "text",
      "title": i18n.__('title_turkish'),
      "payload": 'Turkish'
    },
    {
      "content_type": "text",
      "title": i18n.__('title_polish'),
      "payload": 'Polish'
    },
    {
      "content_type": "text",
      "title": i18n.__('title_italian'),
      "payload": 'Italian'
    },
    {
      "content_type": "text",
      "title": i18n.__('title_romanian'),
      "payload": 'Romanian'
    }
  ];

  let guessedNationality = convo.vars.nationality || convo.vars.passport_issuer || "";

  if( guessedNationality ) {
    // Use the proper nationality name
    if( typeof nationalities[guessedNationality.toLowerCase()] === 'string' ) {
      let nationalityName = nationalities[guessedNationality.toLowerCase()]
      nationalityName = nationalityName[0].toUpperCase() + nationalityName.slice(1)

      commonNationalities.unshift({
        "content_type": "text",
        "title": nationalityName,
        "payload": nationalityName
      })
    }
  }

  convo.ask (
  {
      text: i18n.__('ask_nationality'),
    quick_replies: commonNationalities
  }, [{
    default: true,
    callback: function (response, convo) {
        logger.info("=======enhlish resonse ",response)
      let nationality = response.text;


      logger.debug("&&& natinalite choose 1" , nationality )

      googleMapsClient.geocode({
        address: nationality
      }, function(err, res) {
          logger.debug("&&& natinalite choose 2" , nationality )
          if (!err && res.json.status === "OK") {
            convo.setVar('nationality', nationality)
              logger.debug("&&& natinalite choose 3 ok" , nationality )
            let countryDetail = i18n.__('reply_country_details')
            switch(response.text.toLowerCase()) {
              case 'turkish':
                  countryDetail = i18n.__('reply_country_details_turkish')
                break;
              case 'italian':
                  countryDetail = i18n.__('reply_country_details_italian')
                break;
              case 'spanish':
                  countryDetail = i18n.__('reply_country_details_spanish')
                break;
              case 'argentinian':
                  countryDetail = i18n.__('reply_country_details_argentinian')
                break;

            }
            convo.say(countryDetail)
            typeof next === 'function' ? next(response, convo) : null;
            convo.next();
            } else if (res.json.status === "ZERO_RESULTS") {
              logger.debug("&&& natinalite choose 4 zerrlo" , nationality )
              logger.debug("status is "+ res.json.status);
              convo.say(i18n.__('error_message_wrong_nationality_input'));
              convo.repeat();
              convo.next();
              return
          }
        })

      }
    } ]
        )
```

**Comment by dale-french** [comment]
Same here

```
module.exports = function (controller) {

    controller.hears(['dinner'], 'message_received', function (bot, message) {

        bot.reply(message, `Here's some of this weeks recipes`);
        // Show the carousel of recipes
        bot.reply(message, responses.recipesCarousel(recipes));

    });

};
```

**Comment by swagat123** [comment]
Same here , i faced the issue .

An error occured in the receive middleware:  TypeError: Cannot read property 'name' of undefined
    at patterns.some.pattern (/home/ec2-user/misc/middleware-rasa.js:43:28)

Is there any update on this.

**Comment by artur79** [comment]
That error may be triggered when you call some method with wrong params, for example in my case I've called text.includes(/foo/) instead _text.includes('foo')_ inside controller.hears callback

**Comment by ventz** [comment]
@artur79 There seems to be something else going on here because even an empty controller.hears generates this:

```
module.exports = function(controller) {

    controller.hears(['test'], 'ambient', function (bot, message) {
    });
}
```

The main section is loading the module via:

```
var normalizedPath = require("path").join(__dirname, "skills");
require("fs").readdirSync(normalizedPath).forEach(function(file) {
    require("./skills/" + file)(controller);
});
```

**Comment by ventz** [comment]
On the upside -- it seems the just released version `0.6.20` fixed that issue! :)
